citesphere/src/test/java/edu/asu/diging/citesphere/core/service/impl/CitationManagerTest.java

```diff
@@ -37,217 +40,303 @@
         managerToTest.updateCitation(user, GROUP_ID, existingCitation);
     }
-    
+
+    @Test(expected = GroupDoesNotExistException.class)
+    public void test_deleteCitation_groupDoesNotExist() throws ZoteroConnectionException, GroupDoesNotExistException,
+            CitationIsOutdatedException, ZoteroHttpStatusException {
+
+        Mockito.when(groupRepository.findById(new Long(GROUP_ID))).thenReturn(Optional.empty());
+
+        managerToTest.deleteCitation(user, GROUP_ID, existingCitation);
+    }
+
+    @Test(expected = CitationIsOutdatedException.class)
+    public void test_deleteCitation_citationOutdated() throws ZoteroConnectionException, GroupDoesNotExistException,
+            CitationIsOutdatedException, ZoteroHttpStatusException {
+        Long group3Id = new Long(GROUP_ID);
+        CitationGroup group = new CitationGroup();
+        group.setId(group3Id);
+        group.setVersion(new Long(2));
+        Optional<CitationGroup> optionalGroup = Optional.of(group);
+
+        Mockito.when(groupRepository.findById(new Long(GROUP_ID))).thenReturn(optionalGroup);
+        Mockito.when(zoteroManager.getGroupItemVersion(user, GROUP_ID, existingCitation.getKey()))
+                .thenReturn(new Long(2));
+        Mockito.when(citationRepository.findById(existingCitation.getKey())).thenReturn(Optional.of(existingCitation));
+        managerToTest.deleteCitation(user, GROUP_ID, existingCitation);
+    }
+
+    @Test
+    public void test_deleteCitation_success() throws ZoteroConnectionException, GroupDoesNotExistException,
+            ZoteroHttpStatusException, CitationIsOutdatedException {
+
+        Long group3Id = new Long(GROUP_ID);
+        CitationGroup group = new CitationGroup();
+        group.setId(group3Id);
+        group.setVersion(new Long(2));
+        Optional<CitationGroup> optionalGroup = Optional.of(group);
+
+        List<ICitation> citations = new ArrayList<>();
+        citations.add(existingCitation);
+
+        List<PageRequest> requests = new ArrayList<PageRequest>();
+        PageRequest request = new PageRequest();
+        request.setCitations(citations);
+        requests.add(request);
+
+        Mockito.when(zoteroManager.getGroupItemVersion(user, GROUP_ID, existingCitation.getKey()))
+                .thenReturn(currentVersion);
+        Mockito.when(citationRepository.findById(existingCitation.getKey())).thenReturn(Optional.of(existingCitation));
+        Mockito.when(groupRepository.findById(new Long(GROUP_ID))).thenReturn(optionalGroup);
+        Mockito.when(pageRequestRepository.findByCitations(existingCitation)).thenReturn(requests);
+
+        managerToTest.deleteCitation(user, GROUP_ID, existingCitation);
+
+        Mockito.verify(zoteroManager).deleteCitation(user, GROUP_ID, existingCitation);
+        Mockito.verify(citationRepository).delete((Citation) existingCitation);
+
+        Assert.assertTrue(requests.get(0).getCitations().size() == 0);
```

Comment:
let's use `assertEquals` so the if this fails the error message will state what was expected and what was the value.